src/PlaybackLooping.cpp

```diff
@@ -0,0 +1,208 @@
+/*!********************************************************************
+*
+ Audacity: A Digital Audio Editor
+
+ PlaybackLooping.cpp
+
+ Anton Gerasimov
+
+ **********************************************************************/
+
+#include "PlaybackLooping.h"
+
+#include "Theme.h"
+#include "AllThemeResources.h"
+
+#include "ZoomInfo.h"
+#include "HitTestResult.h"
+#include "RefreshCode.h"
+#include "TrackPanelMouseEvent.h"
+#include "SelectUtilities.h"
+#include "Track.h"
+
+#include <wx/event.h>
+
+PlaybackLooping::PlaybackLooping()
+{
+	;
+}
+
+PlaybackLooping& PlaybackLooping::GetInstance()
+{
+	static PlaybackLooping instance;
+
+	return instance;
+}
+
+bool PlaybackLooping::IsLoopingEnabled() const
+{
+	return mLoopingEnabled;
+}
+
+void PlaybackLooping::EnableLooping(bool enable)
+{
+	mLoopingEnabled = enable;
+}
+
+bool PlaybackLooping::IsProcessing() const
+{
+	return mProcessingState != ProcessingStates::kIdle;
+}
+
+bool PlaybackLooping::LoopingProcessor(AudacityProject* pProject, wxCoord& mousePosX, ProcessingEvents processingEvents)
+{
+	// This is empty project.
+	if (TrackList::Get(*pProject).size() == 0) // NOTE: need check on record?
+		return false;
+
+	switch (mProcessingState)
+	{
+	case ProcessingStates::kIdle:
+		if (processingEvents == ProcessingEvents::kClick)
+		{
+			const auto& viewInfo = ViewInfo::Get(*pProject);
+			const auto& playRegion = viewInfo.playRegion;
+			mOldPlayRegion = playRegion;
+			//SelectUtilities::LockPlayRegion(*pProject);
+			EnableLooping(true);
+
+			auto loopingPos = viewInfo.PositionToTime(mousePosX, viewInfo.GetLeftOffset(), false);
+			mLoopedPlayRegion.SetStart(loopingPos);
+
+			mProcessingState = ProcessingStates::kEnterLooping;
+		}
+		break;
+
+	case ProcessingStates::kEnterLooping:
+		if (processingEvents == ProcessingEvents::kDrag)
+		{
```

Comment:
This incomplete state machine (which as I said, you should not need) is broken.  It never enters the drag state.

I can't actually drag the play region in this build.

I thought the intention also was different dragging behavior, in case of clicking inside the play region (to slide it as a whole), or at an edge (to stretch or shorten it).  None of that yet here.